x-pack/spec/filters/geoip/database_metadata_spec.rb

```diff
@@ -50,14 +50,23 @@ module LogStash module Filters module Geoip
     end
 
     context "save timestamp" do
+      before do
+        ::File.open(DEFAULT_CITY_GZ_PATH, "w") { |f| f.write "make a non empty file" }
+      end
+
+      after do
+        delete_file(DEFAULT_CITY_GZ_PATH)
+      end
+
       it "write the current time" do
         dbm.save_timestamp(DEFAULT_CITY_DB_PATH)
 
         metadata = dbm.get_metadata.last
         expect(metadata[DatabaseMetadata::Column::DATABASE_TYPE]).to eq("City")
         past = metadata[DatabaseMetadata::Column::UPDATE_AT]
         expect(Time.now.to_i - past.to_i).to be < 100
-        expect(metadata[DatabaseMetadata::Column::GZ_MD5]).to eq('')
+        expect(metadata[DatabaseMetadata::Column::GZ_MD5]).not_to be_empty
+        expect(metadata[DatabaseMetadata::Column::GZ_MD5]).to eq(md5(DEFAULT_CITY_GZ_PATH))
```

Comment:
~I'm having a hard time finding where this `md5` method is coming from, and am worried that we are polluting the `Object` space somehow in these specs in a way that is not replicated in released artifacts.~ _EDIT: we are in fact polluting `Object`, but not in the way I had guessed. See other comment about our spec helpers._

I see that this spec opens up the Geoip filter's definition, then opens up a spec context with `describe`, effectively monkey-patching the filter definition in the process. This is at best unsafe, and at worst will cause us to validate behaviour provided by the specs themselves instead of code available from lib.

~~~ ruby
module LogStash module Filters module Geoip
  describe DatabaseMetadata, :aggregate_failures do
    # ...
  end
end end end
~~~

Should be:

~~~ ruby
context LogStash::Filters::Geoip do
  describe DatabaseMetadata, :aggregate_failures do
    # ...
  end
end

~~~